### PR TITLE
Clean up bukkit listeners by splitting them up

### PR DESCRIPTION
**Pull request by dordsor21**
Shouldn't have broken anything.... Probably just worth someone else double-checking the listeners are definitely called

### CODE REVIEW

**Comment by N0tMyFaultOG** [comment]
You forgot a few license headers.

File: Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java [hunk 1]
```diff
@@ -0,0 +1,1075 @@
...
+
+        // Check liquid flow flag inside of origin plot too
+        final Location fLocation = BukkitUtil.getLocation(from.getLocation());
+        final PlotArea fromArea = fLocation.getPlotArea();
+        if (fromArea != null) {
+            final Plot plot = fromArea.getOwnedPlot(fLocation);
+            if (plot != null
+                && plot.getFlag(LiquidFlowFlag.class) == LiquidFlowFlag.FlowStatus.DISABLED && event
+                .getBlock().isLiquid()) {
+                plot.debug("Liquid could now flow because liquid-flow = disabled");
```
Comment (N0tMyFaultOG) [inline Bukkit/src/main/java/com/plotsquared/bukkit/listener/BlockEventListener.java:664]: Little typo here

File: Bukkit/src/main/java/com/plotsquared/bukkit/listener/EntityEventListener.java [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+        Location location = BukkitUtil.getLocation(block.getLocation());
+        PlotArea area = location.getPlotArea();
+        if (area == null) {
+            return;
+        }
+        Plot plot = area.getOwnedPlotAbs(location);
+        if (plot == null || plot.getFlag(DisablePhysicsFlag.class)) {
+            event.setCancelled(true);
+            if (plot != null) {
+                plot.debug("Fallin block event was cancelled because disable-physics = true");
```
Comment (N0tMyFaultOG) [inline Bukkit/src/main/java/com/plotsquared/bukkit/listener/EntityEventListener.java:203]: Typo

File: Bukkit/src/main/java/com/plotsquared/bukkit/listener/ProjectileEventListener.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        }
+        PlotPlayer<Player> pp = BukkitUtil.getPlayer((Player) shooter);
+        Plot plot = location.getOwnedPlot();
+        if (plot != null && !plot.isAdded(pp.getUUID())) {
+            entity.remove();
+            event.setCancelled(true);
+        }
+    }
+
+    @SuppressWarnings({"BooleanMethodIsAlwaysInverted", "cos it's not... dum IntelliJ"}) @EventHandler
```
Comment (N0tMyFaultOG) [inline Bukkit/src/main/java/com/plotsquared/bukkit/listener/ProjectileEventListener.java:108]: cute

**Comment by N0tMyFaultOG** [comment]
I have given this branch a test and noticed a few things:
- Unable to place armor stands or mobs on roads and unowned plots
- misc and entity caps are no longer respected. I dropped a few netherite chestplates but it easily goes over the limit 
Other entities like armor stands are fine

**Comment by dordsor21** [comment]
> I have given this branch a test and noticed a few things:
> 
> * Unable to place armor stands or mobs on roads and unowned plots
> * misc and entity caps are no longer respected. I dropped a few netherite chestplates but it easily goes over the limit 
>   Other entities like armor stands are fine

All I did was copy-paste what was there. No behaviour was changed anywhere. If it's an issue here, it'll be an issue on main branch.

**Comment by N0tMyFaultOG** [comment]
> All I did was copy-paste what was there. No behaviour was changed anywhere. If it's an issue here, it'll be an issue on main branch.

Looks like something that hasn't been addressed then.

**Pull request merged**